### PR TITLE
Accent trait limit

### LINKED ISSUE #20501: should be traits limit
why?
it's very weird speech, when selecting all accent traits
OR
make limit only to accent traits

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
updated traits code, removed obsolete constructs. Traits are now divided into categories, where you can customize how many traits you can take in a category.

Also added Italian and cowboy accent traits.

## Why / Balance
5 accents on one character makes speech horribly unintelligible.
resolve #20501
## Media
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Added italian and cowboy accents
- fix: Now you can't choose all the accents at once.

### CODE REVIEW

**Comment by lzk228** [comment]
resolves #20501

**Comment by TheShuEd** [comment]
image

**Comment by TheShuEd** [comment]
wait im fixing some bugs

**Comment by TheShuEd** [comment]
Frontal Lisp is now worth 2 points

File: Content.Client/Lobby/UI/HumanoidProfileEditor.xaml.cs [hunk 1]
```diff
@@ -463,41 +464,65 @@ public void RefreshTraits()
         {
             TraitsList.DisposeAllChildren();
 
-            var traits = _prototypeManager.EnumeratePrototypes<TraitPrototype>().OrderBy(t => Loc.GetString(t.Name)).ToList();
+            var categories = _prototypeManager.EnumeratePrototypes<TraitCategoryPrototype>().OrderBy(t => Loc.GetString(t.Name)).ToList();
```
Comment (Tornado-Technology) [inline Content.Client/Lobby/UI/HumanoidProfileEditor.xaml.cs:467]: Scary Linq

Comment (TheShuEd) [reply Content.Client/Lobby/UI/HumanoidProfileEditor.xaml.cs:467]: image
there was a scary linq here before me.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
currently there seems to be an issue if a trait doesn't end up in a category. I'd say make it so either:
1. there's a test that errors if a trait isn't in any category
2. trait prototypes have the category themselves, instead of larger lists (This is better for adding new categories and traits in the future i think)
3. Add some kind of 'default' category that always shows up in the UI.

other than that it looks good

**Comment by Emisse** [comment]
agree with emo, needs to automatically add it to the category. we have a lot of stuff we need to consolidate to not be hardcoded into lists and i dont feel comfortable merging something that adds another one

**Comment by TheShuEd** [comment]
now the category prototype does not contain the trait prototypes themselves:
```yml
- type: traitCategory
  id: Disabilities
  name: trait-category-disabilities

- type: traitCategory
  id: SpeechTraits
  name: trait-category-speech
  maxTraitPoints: 2
  ```
but the category is written inside the trait itself. If no category is specified, the trait will be displayed in the UI above all categories, without a category header.
```yml
- type: trait
  id: ItalianAccent
  name: trait-italian-name
  description: trait-italian-desc
  category: SpeechTraits
  cost: 1
  components:
    - type: ReplacementAccent
      accent: italian

- type: trait
  id: Liar
  name: trait-liar-name
  description: trait-liar-desc
  category: SpeechTraits
  cost: 1
  components:
    - type: ReplacementAccent
      accent: liar
```

**Pull request merged**